### PR TITLE
PLATFORM-783: get the list of wikis where the given article exists

### PR DESCRIPTION
**Pull request by macbre**
https://wikia-inc.atlassian.net/browse/PLATFORM-783

`MultiTaskDelete`, when not provided with the list of wikis, loops on all active wikis and executes the maintenance script.

This pull request changes this behavior. We now get the list of wikis where the given article exists. This will greatly speed up the `MultiTaskDelete` execution and avoid the issue with deploy tools garbage collecting the old checkouts causing long running task to fail to execute maintenance scripts.

``` sql
>select * from pages where page_title_lower = 'rynek_jeżycki' and page_namespace = 0;
+---------------+---------+----------------+------------------+----------------+-------------+-----------------+------------------+------------+-------------+---------------------+-------------------+
| page_wikia_id | page_id | page_namespace | page_title_lower | page_title     | page_status | page_is_content | page_is_redirect | page_edits | page_latest | page_last_edited    | page_last_indexed |
+---------------+---------+----------------+------------------+----------------+-------------+-----------------+------------------+------------+-------------+---------------------+-------------------+
|          5915 |   29445 |              0 | rynek_jeżycki   | Rynek_Jeżycki |           0 |               1 |                0 |          2 |       96155 | 2015-02-26 17:57:16 |              NULL |
+---------------+---------+----------------+------------------+----------------+-------------+-----------------+------------------+------------+-------------+---------------------+-------------------+
1 row in set (0.01 sec)

>explain select * from pages where page_title_lower = 'tomyvilu' and page_namespace = 2;
+----+-------------+-------+------+-----------------------------------------------+------------------------+---------+-------+------+-------------+
| id | select_type | table | type | possible_keys                                 | key                    | key_len | ref   | rows | Extra       |
+----+-------------+-------+------+-----------------------------------------------+------------------------+---------+-------+------+-------------+
|  1 | SIMPLE      | pages | ref  | page_title_lower_wikia,page_title_lower_edits | page_title_lower_wikia | 258     | const |   47 | Using where |
+----+-------------+-------+------+-----------------------------------------------+------------------------+---------+-------+------+-------------+
1 row in set (0.00 sec)
```

``` php
> print_r(Wikia\Tasks\Tasks\MultiTask::getWikisWherePageExists(Title::newFromText('Forum')))
Array
(
    [0] => 15
    [1] => 19
    [2] => 32
    [3] => 38
    [4] => 51
    [5] => 58
    [6] => 71
    [7] => 74
    [8] => 84
    [9] => 95
...
)
```

The fix itself is in  709a628

@michalroszka / @wladekb 

/cc @Wikia/community-engineering

### CODE REVIEW

File: includes/wikia/tasks/Tasks/MultiTask.class.php [hunk 1]
```diff
@@ -159,38 +165,92 @@ public function edit( $params ) {
...
+	 */
+	private function getWikisWherePageExists( \Title $title ) {
+		global $wgExternalDatawareDB;
+
+		$dbr = wfGetDB( DB_SLAVE, [], $wgExternalDatawareDB );
+		$res = $dbr->select(
+			'pages',
+			'page_wikia_id as wiki_id',
+			[
+				'page_title_lower' => str_replace( ' ', '_', mb_strtolower( $title->getText() ) ),
```
Comment (Grunny) [inline includes/wikia/tasks/Tasks/MultiTask.class.php]: Any reason not to just use `->getDBkey()` instead doing the replace on `getText()`?

Comment (macbre) [inline includes/wikia/tasks/Tasks/MultiTask.class.php]: Not really, fixed. Thanks!

**Comment by Grunny** [comment]
:+1: LGTM. :)

File: includes/wikia/tasks/Tasks/MultiTask.class.php [hunk 2]
```diff
@@ -159,38 +165,92 @@ public function edit( $params ) {
...
 				}
 			}
 		}
-		
+
+		if ( !is_array( $wikisId ) ) {
+			$wikisId = [ $wikisId ];
+		}
+
+		$this->info( sprintf( 'Running "%s" action of %d wiki(s)', $action, count( $wikisId ) ), [
```
Comment (wladekb) [inline includes/wikia/tasks/Tasks/MultiTask.class.php]: s/of/on/

File: includes/wikia/tasks/Tasks/MultiTask.class.php [hunk 3]
```diff
@@ -199,19 +259,19 @@ private function runOnWikis($wikiInputRaw, $lang, $cat, $wikiId, $action, $comma
 			$query->AND_( 'city_list.city_lang' )->EQUAL_TO( $lang );
 		}
 
-		if (!empty($wikiId)) {
-			$query->AND_('city_list.city_id')->EQUAL_TO($wikiId);
+		if ( !empty( $wikiId ) ) {
```
Comment (wladekb) [inline includes/wikia/tasks/Tasks/MultiTask.class.php]: 1. s/wikiId/wikisId/

2. This entire PR purpose is to drop support for processing all wikis. Maybe we should fail early here with a descriptive error message if we discover $wikisId is empty?

Comment (macbre) [inline includes/wikia/tasks/Tasks/MultiTask.class.php]: 2. True, but I want to start with delete task. Edit and move can still be executed with no wiki selected. Once we prove this solution works on production, I'll modify the remaining task types.

**Pull request merged**